### PR TITLE
Before sending to Sentry check if the error is a known chain error

### PR DESCRIPTION
**Pull request by andresgutgon**
# What?
We're receiving in Sentry errors that are not bugs. Only errors on users' prompts. Which we already store for then in logs to check.

## Sentry
https://latitude-l5.sentry.io/issues/6011193645/events/e3c32a7ace82448ab0bcb9b11d8df436/events/?fov=0%2C413000&project=4507922531418112&referrer=issue_details.related_trace_issue

## Off-topic error
```
Only plain objects, and a few built-ins, can be passed to Client Components from Server Components. Classes or null prototypes are not supported.
  {}
```
https://latitude-l5.sentry.io/issues/6008744834/events/0fc792f692404c558898ce7919e0ac66/events/?project=4507922531418112

### CODE REVIEW

File: apps/web/src/actions/prompts/run.ts [hunk 1]
```diff
@@ -35,7 +35,6 @@ export const runPromptAction = authProcedure
 
       return {
         output: stream.value,
-        response: run.response,
```
Comment (andresgutgon) [inline apps/web/src/actions/prompts/run.ts:38]: This fix this error
https://latitude-l5.sentry.io/issues/6008744834/events/827e54994bab428dbff8b7c5df8f8e40/?project=4507922531418112

File: apps/gateway/src/routes/api/v1/projects/[projectId]/versions/[versionUuid]/documents/handlers/run.test.ts [hunk 2]
```diff
@@ -18,7 +19,6 @@ import { apiKeys } from '@latitude-data/core/schema'
 import { mergeCommit } from '@latitude-data/core/services/commits/merge'
 import { parseSSEvent } from '$/common/parseSSEEvent'
 import app from '$/routes/app'
-import { eq } from 'drizzle-orm'
```
Comment (andresgutgon) [inline apps/gateway/src/routes/api/v1/projects/[projectId]/versions/[versionUuid]/documents/handlers/run.test.ts:21]: This was making the test fail in my machine. Drizzle from an app fails. It has to be run from core

File: packages/core/src/services/chains/ChainErrors/index.ts [hunk 3]
```diff
@@ -1,9 +1,10 @@
 import { RunError } from '../../../browser'
 import { RunErrorCodes, RunErrorDetails } from '../../../constants'
+import { LatitudeError, LatitudeErrorDetails } from '../../../lib'
 
-export class ChainError<T extends RunErrorCodes> extends Error {
+export class ChainError<T extends RunErrorCodes> extends LatitudeError {
```
Comment (andresgutgon) [inline packages/core/src/services/chains/ChainErrors/index.ts:5]: I'm inheriting from `LatitudeError` so when failing in the gateway users see this output
https://github.com/latitude-dev/latitude-llm/blob/main/apps/gateway/src/middlewares/errorHandler.ts#L30-L34

What do you think?

**Review by csansoon — DISMISSED** [review]

**Comment by andresgutgon** [comment]
Ignoring the error in Sentry is not solving the underlaying issue. We're throwing `NotFound` errors when not user. What we need to do is redirect to `/login`. We need to think in the more nextjs way of doing this.

**Review by csansoon — APPROVED** [review]

File: apps/gateway/src/middlewares/errorHandler.ts [hunk 4]
```diff
@@ -4,12 +4,17 @@ import {
 } from '@latitude-data/core/lib/errors'
 import { captureException } from '$/common/sentry'
 import { HTTPException } from 'hono/http-exception'
+import { getUnknownError } from 'node_modules/@latitude-data/core/src/jobs/utils/getUnknownError'
```
Comment (geclos) [inline apps/gateway/src/middlewares/errorHandler.ts:7]: weird import

File: apps/gateway/src/middlewares/errorHandler.ts [hunk 5]
```diff
@@ -4,12 +4,17 @@ import {
...
 import HttpStatusCodes from '../common/httpStatusCodes'
 
 const errorHandlerMiddleware = (err: Error) => {
   if (process.env.NODE_ENV !== 'test') {
-    captureException(err)
+    const unknownError = getUnknownError(err)
+
+    if (unknownError) {
+      captureException(err)
+    }
```
Comment (geclos) [inline apps/gateway/src/middlewares/errorHandler.ts:17]: we don't send the exception if we don't identify the user?

File: apps/gateway/src/routes/api/v1/conversations/[conversationUuid]/handlers/chat.test.ts [hunk 6]
```diff
@@ -94,10 +92,9 @@ describe('POST /add-message', () => {
 
       const { workspace: wsp } = await createProject()
       workspace = wsp
-      // TODO: move to core
```
Comment (geclos) [inline apps/gateway/src/routes/api/v1/conversations/[conversationUuid]/handlers/chat.test.ts:97]: thanks!

File: apps/gateway/src/routes/api/v1/projects/[projectId]/versions/[versionUuid]/documents/handlers/run.ts [hunk 7]
```diff
@@ -5,6 +5,7 @@ import { runDocumentAtCommit } from '@latitude-data/core/services/commits/runDoc
 import { captureException } from '$/common/sentry'
 import { Factory } from 'hono/factory'
 import { streamSSE } from 'hono/streaming'
+import { getUnknownError } from 'node_modules/@latitude-data/core/src/jobs/utils/getUnknownError'
```
Comment (geclos) [inline apps/gateway/src/routes/api/v1/projects/[projectId]/versions/[versionUuid]/documents/handlers/run.ts:8]: weird import

File: packages/core/src/jobs/utils/getUnknownError.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * We only throw an error if is not a known run error.
+ *
+ *  1. Is a ChainError of type `unknown`. This is the catch in `runChain` not knowing what happened.
+ *  2. is not a `ChainError` so it means something exploded.
+ */
+export function getUnknownError(error: Error | unknown | undefined) {
+  const isAllGood =
+    !error || (isChainError(error) && error.errorCode !== RunErrorCodes.Unknown)
```
Comment (geclos) [inline packages/core/src/jobs/utils/getUnknownError.ts:12]: aaah ok

**Review by geclos — DISMISSED** [review]
fix the import path and 🚀

File: packages/core/src/jobs/job-definitions/batchEvaluations/runEvaluationJob.ts [hunk 9]
```diff
@@ -107,5 +89,7 @@ export async function runEvaluationJob(job: Job<RunEvaluationJobData>) {
     },
   })
 
-  throwIfUnknownError(error)
+  const unknownError = getUnknownError(error)
+
+  if (unknownError) throw unknownError
```
[the remark this answers is not in the document]
Comment (andresgutgon) [reply packages/core/src/jobs/job-definitions/batchEvaluations/runEvaluationJob.ts:94]: In the others, I'm not throwing. I had this before here

File: apps/gateway/src/routes/api/v1/projects/[projectId]/versions/[versionUuid]/documents/handlers/run.ts (hunk 7 above)
Comment (andresgutgon) [reply apps/gateway/src/routes/api/v1/projects/[projectId]/versions/[versionUuid]/documents/handlers/run.ts:8]: What do you mean?

Comment (andresgutgon) [reply apps/gateway/src/routes/api/v1/projects/[projectId]/versions/[versionUuid]/documents/handlers/run.ts:8]: I see

**Pull request merged**